terraform/services/variables.tf

```diff
@@ -13,3 +13,27 @@ variable "image" {
 variable "storage_pool" {
   type = string
 }
+
+variable "nic_method" {
+  type    = string
+  default = "network"
+  validation {
+    condition     = contains(["network", "nictype"], var.nic_method)
+    error_message = "Valid value is one of the following: network, nictype."
+  }
+}
+
+variable "nictype" {
+  type    = string
+  default = ""
+}
+
+variable "nictype_parent" {
+  type    = string
+  default = ""
+}
+
+variable "network_name" {
+  type    = string
+  default = ""
+}
```

Comment:
What's the use case for this?

The example used is one where the default use of `network` is better suited.
If this is to support a pre-existing bridge or macvlan, I think it'd be better for the user to use `incus network create SOME-NETWORK --type=physical parent=EXISTING-INTERFACE` and use the resulting network rather than use `nictype` to directly get to the NIC.